### PR TITLE
feat: add documenation for Katsu

### PR DESCRIPTION
**Pull request by lleyton**
Imported and modified from https://github.com/FyraLabs/devdocs-old/pull/8/

### CODE REVIEW

File: pages/katsu/configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+```yaml
+distro: Fedora 39
+```
+
+### `scripts`
+
+This section defines the scripts that will be run before and after the image is built.
+
+They have two sub-sections, `pre` and `post`, which define the scripts to run before and after the image is built
+Both of these has the same format, described below
```
Comment (madonuko) [inline pages/katsu/configuration.mdx]:
```suggestion
They have two sub-sections, `pre` and `post`, which define the scripts to run before and after the image is built.
Both of these has the same format described below.
```

File: pages/katsu/configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+The filesystem to format the partition with `mkfs`
+
+##### `disk.partitions.[].mountpoint`
+
+The mountpoint to mount the partition at relative to the root of the image.
+
+---
+
+The order of the partitions is important, as they are created in the order they are defined, however, mountpoints will be sorted
+by nesting level, so `/` will be mounted first, then `/boot`, then `/boot/efi` This is done using a topological sort algorithm.
```
Comment (madonuko) [inline pages/katsu/configuration.mdx]:
```suggestion
by nesting level, so `/` will be mounted first, then `/boot`, then `/boot/efi`. This is done using a topological sort algorithm.
```

File: pages/katsu/configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+If the mountpoint depths are the same, they will be sorted alphabetically.
+
+---
+
+### `import.[]`
```
Comment (madonuko) [inline pages/katsu/configuration.mdx:190]: Might want to also state that not all attributes are going to be merged / have the list here

https://github.com/FyraLabs/katsu/blob/main/src/config.rs#L154

**Review by madonuko — CHANGES_REQUESTED** [review]

File: pages/katsu/configuration.mdx (hunk 3 above)
Comment (lleyton) [reply pages/katsu/configuration.mdx:190]: Look better?

**Review by madonuko — APPROVED** [review]
:3

**Pull request merged**